### PR TITLE
[ci] Add workflow to automatically create a GitHub release

### PR DESCRIPTION
**Pull request by Drarig29**
- #128
- #129 ← 
---

This PR adds a workflow that will be automatically run when a release PR is merged.

### CODE REVIEW

File: .github/workflows/release-version-on-merge.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  pull_request:
+    types:
+      - closed
+
+permissions:
+  contents: write
+
+jobs:
+  release-version-on-merge:
+    if: github.event.pull_request.merged == true && startsWith(github.event.pull_request.title, '[release:')
```
Comment (etnbrd) [inline .github/workflows/release-version-on-merge.yml:12]: This means the release PR will need to have their title prefixed with `[release:`, right? Is that documented somewhere?

File: .github/workflows/release-version-on-merge.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create GitHub release
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const tagName = '${{ github.event.pull_request.head.ref }}'.replace('release/', '')
```
Comment (etnbrd) [inline .github/workflows/release-version-on-merge.yml:19]: This means the release branch will need to be prefixed with `release/` and only contain the tag name, right? Is that documented somewhere?

**Review by etnbrd — COMMENTED** [review]
If my understanding is correct, when a branch named `release/<major-minor-patch>` is merged by a PR prefixed with `[release:`, then this workflow will create a github release named and tagged after the `major-minor-patch` in the branch name. Is that correct?

**Review by etnbrd — APPROVED** [review]
LGTM 👍

**Comment by Drarig29** [comment]
I'll use the new authentication system in another PR.

**Pull request merged**